cpp/ql/lib/semmle/code/cpp/commons/Exclusions.qll

```diff
@@ -109,3 +109,39 @@ predicate isFromMacroDefinition(Element e) {
     eLocation.getEndColumn() >= miLocation.getEndColumn()
   )
 }
+
+/**
+ * Holds if `e` is completely or partially from a macro definition, as opposed
+ * to being passed in as an argument.
+ *
+ * In the following example, the call to `f` is from a macro definition,
+ * while `y`, `+`, `1`, and `;` are not. This assumes that no identifier apart
+ * from `M` refers to a macro.
+ * ```
+ * #define M(x) f(x)
+ * ...
+ *   M(y + 1);
+ * ```
+ */
+predicate isFromMacroDefinition(Element e) {
+  isFromMacroInvocation(e, _)
+}
+
+/**
+ * Holds if `e` is completely or partially from a _system macro_ definition, as
+ * opposed to being passed in as an argument. A system macro is a macro whose
+ * definition is outside the source directory of the database.
+ * 
+ * If the system macro is invoked through a non-system macro, then this
+ * predicate does not hold. That's a limitation of how macros are represented
+ * in the database.
```

Comment:
Could you share an example?  (I'd have thought between `getAnExpandedElement()` and `MacroInvocation.getParentInvocation()` we can work this out, I could be wrong)

Reply:
I've pushed a commit that removes the claim about db limitations.